src/stampit.js

```diff
@@ -124,6 +132,16 @@ function compose(...stamps) {
     stamps.push(this);
   }
 
+  /*
+   * If stamp has unique displayName and is cached, return it. A composed stamp
+   * should have a unique displayName. Since the compose method overrides with last-in
+   * priority, the displayName should be at the end.
+   */
+  let displayName = stamps[stamps.length - 1].displayName || 'ReactStamp';
```

Comment:
I don't think I should be defaulting to 'ReactStamp' here. What if the `displayName` prop is found in a stamp at the beginning of the composition chain?